changelog-tool/main.pony

```diff
@@ -171,6 +182,24 @@ actor Main
       err("unable to perform release preparation")
     end
 
+  fun cmd_add(
+    filepath: FilePath,
+    filename: String,
+    section: String,
+    entry: String,
+    edit: Bool)
+  =>
+    try
+      edit_or_print(
+        filepath,
+        edit,
+        Changelog(parse(filepath, filename)?)?
```

Comment:
Adding `.> create_unreleased()` here before `add_entry` will likely solve the problem. An unreleased section should be created anyway if someone is using this command.

Reply:
Hmm, creating a new "unreleased" release creates new empty sections.

```bash
changelog-tool new
changelog-tool unreleased -e
changelog-tool add fixed "We fixed some nasty bug." -e
changelog-tool add added "And we added some cool feature." -e
changelog-tool add changed "And changed some little things as well." -e
changelog-tool release 1.0.0
```

Currently, this works as expected. If I add that command only the last "changed" will be reflected in the file, right?

Reply:
Yeah, the problem is that the "unreleased" release must contain all sections if it exists. Previously the `string` function would assume that it is empty if it exists. But this is no longer a reasonable assumption, so the sections must be created if they don't exist. That's why it may be useful to create a new class for unreleased that reflects the requirement that the sections cannot have type `None`.

Reply:
Makes sense, make the types do the work instead.